optuna/samplers/_tpe/sampler.py

```diff
@@ -192,6 +258,18 @@ def __init__(
                 ExperimentalWarning,
             )
 
+        if group:
+            if not multivariate:
+                raise ValueError(
+                    "``group`` option can only be enabled when ``multivariate`` is " "enabled."
+                )
+            else:
+                warnings.warn(
```

Comment:
Not a strong opinion but I think you can raise this warning independent of `multivariate`.